lib/Settings/Admin.php

```diff
@@ -57,62 +57,65 @@ public function getForm(): TemplateResponse {
 		$mounts_json = $this->config->getAppValue($this->appName,"storageconfig","");	//default
 		$mounts = json_decode($mounts_json,  true);
```

Comment:
When nothing is configured the `mounts_json` variable is set to an empty string. When this is run through `json_decode` then `null` is returned. So right below when you check if it's an array, that is of course false, but that also means on a brand new install you get stuck with the `Please configure a Swarm storage in the "External Storage" Administration section` message, even after you have configured a swarm storage.

Reply:
Resolved